### ISSUE TITLE
Armstate Ammo

### ISSUE THREAD

**Issue by Status12** [ISeeDEDPpl/Questor#333]
Cannot reload ammo type into ship. Trunk had worked but it was merged with BE. Questor goes to error state. This is the console readout. I have enough ammo in corp hangar and all of my settings are correct. New support DirectEVE from the site doesnt work either.

09:25:49 [AgentInteraction] Done
09:25:50 [Arm] Begin
09:25:52 [Arm] Activating combat ship
09:25:57 [Arm] Missing [6000] units of ammo: [ "typeID" ] with TypeId ["typeID"]
09:25:57 [Arm] Armstate.NotEnoughAmmo

**Comment by Pluckyduck** [comment]
mine can be solved by manually opening the corp hanger.  i think this loads it into cache and then Q can see it.  I had the same thing, I manually opened the hanger and then set Q back to "arm" and it picked up fine.  Still testling, but can we get some kind of, if out of drones or ammo that Q will open corp hangers?

**Comment by ljk1291** [comment]
same problem, decided to try the new version, because 1 of my chars wasnt arming and now they are all failing :\

**Comment by Pluckyduck** [comment]
test this:

start fresh client, let it run to see if it stops at arming state.  IF it does, open the ammo hanger(expand corp hanger) and then move the drop menu from "error" back to "arm".

**Comment by ljk1291** [comment]
@pluckyduck, thanks worked for my tengu pilots, but my marauders seem to have a new problem now.
they just get stuck in a loop opening and closing corgo hold after arming drones :/

16:15:41 [Arm.MoveDrones] neededDrones: 0
16:15:41 [Arm.MoveDrones] MoveItems
16:15:43 [Arm] Closing Fitting Manager Window

**Comment by ljk1291** [comment]
after 30 mins of looping it ended with an error.

16:50:41 [Arm.MoveDrones] neededDrones: 3
16:50:41 [Arm.MoveDrones] Move [ 1 ] Drones into drone bay
16:50:45 [Arm.MoveDrones] neededDrones: 2
16:50:45 [Arm.MoveDrones] Move [ 1 ] Drones into drone bay
16:50:48 [Arm.MoveDrones] neededDrones: 1
16:50:48 [Arm.MoveDrones] Move [ 1 ] Drones into drone bay
16:50:51 [Arm.MoveDrones] neededDrones: 0
16:50:51 [Arm.MoveDrones] MoveItems
16:51:27 [Arm] Armstate.NotEnoughAmmo

**Comment by Pluckyduck** [comment]
i will see if i can repo the bug to find a work around.  I only have one character that uses drones.

**Comment by Pluckyduck** [comment]
go back and make sure that you do not have drones on your fitting and then start it over.  If do have drones on the fitting manager, remove them and test.   I do NOT use the fitting manager at all.

**Comment by ljk1291** [comment]
no i dont have any drones in the fitting.
just got the same error on 1 of my tengus 

00:24:18 [Arm.LoadFitting] Fitting: EM - currentFit: EM
00:24:18 [Arm.LoadFitting] Current fit is now correct
00:24:20 [Arm] Closing Fitting Manager Window
00:24:57 [Arm] Armstate.NotEnoughAmmo

seems to have smth to do with the fitting manager
edit: think i found the problem, questor doesnt seem to be able to find the items you have in <bring> in the mission.xml if you have ^rhem in corp hangar! worked fine before though

**Comment by Pluckyduck** [comment]
update for this issue,

  Q for me at least, does not cache what is in the corp hangers on the first run.  I can solve the problem but manually expanding the corp hanger selection and then selecting "Arm" again and until the client restarts, the problem goes away. 

  Will turn on debug hanger to see about getting a log for it.

**Comment by Pluckyduck** [comment]
http://pastebin.com/EJ7p5tHA

log details with notes

**Comment by ljk1291** [comment]
i just added this to line 408 of combatbehaviours.cs:
Cache.Instance.DirectEve.ExecuteCommand(DirectCmd.OpenCorpHangar);

temporary fix for the problem until ISDP has time to fix it

**Comment by ISeeDEDPpl** [comment]
please test experimental / bleedingedge (at the moment they are the same) and report back.

**Comment by AdditionalFrame** [comment]
Ok. More info on issue, because I've got the same.

If you restart Eve (or just launch), then "Corporation hangars" subtree is folded, even if I set option "Always show full tree". And Questor hangs, because it can't find hangar "Ammo" or whatever. If you'd expand tree mannually and set Questor combat state to "Arm" (it's now "Error" - not enough ammo), then it'll proceed. Until next client start.

I'm using experimental version from Feb 9. Commit hash: 0417ed08aeac2aaba59db4da8df30787ad5ed48c
